src/GafferScene/SceneAlgo.cpp

```diff
@@ -421,7 +421,13 @@ SceneAlgo::History::Ptr historyWalk( const CapturedProcess *process, InternedStr
 
 	for( const auto &p : process->children )
 	{
-		historyWalk( p.get(), scenePlugChildName, parent );
+		// Parents may spawn other processes in support of the requested plug.
+		// We don't want these to show up in history output, so we only include
+		// ones that are directly in service of the requested plug.
+		if( runTimeCast<const ScenePlug>( p->plug->parent() ) && p->plug->getName() == scenePlugChildName )
```

Comment:
`p->plug->parent<ScenePlug>()` is functionally equivalent to but more readable than `runTimeCast<const ScenePlug>( p->plug->parent() )` - might as well use it.

Reply:
Ahh yes, much better - https://github.com/GafferHQ/gaffer/commit/0f9920e5086407f26ed170f2665b0f2556182328. Sorry, forgot about the templating there.